.github/workflows/test.yaml

```diff
@@ -0,0 +1,59 @@
+name: Test
+
+on:
+  - pull_request
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    env:
+      KEYSTORE_COMMON_NAME: keystore-${{ github.sha }}
+      TRUSTSTORE_COMMON_NAME: truststore-${{ github.sha }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: docker/setup-qemu-action@v3
+      - uses: docker/setup-buildx-action@v3
+      - uses: docker/build-push-action@v5
+        with:
+          context: .
+          tags: java-spiffe-helper:test
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - uses: helm/kind-action@v1
+        with:
+          cluster_name: kind
+      - run: kind load docker-image java-spiffe-helper:test --name kind
+      - run: helm upgrade --install -n spire-server spire-crds spire-crds --repo https://spiffe.github.io/helm-charts-hardened/ --create-namespace
+      - run: |
+          helm upgrade --install -n spire-server spire spire \
+            --repo https://spiffe.github.io/helm-charts-hardened/ \
+            -f .github/tests/spire-values.yaml \
+            --set spire-server.ca_subject.common_name="$TRUSTSTORE_COMMON_NAME" \
+            --set spire-server.controllerManager.identities.clusterSPIFFEIDs.java-spiffe-helper.dnsNameTemplates[0]="$KEYSTORE_COMMON_NAME"
+      - run: kubectl apply -f .github/tests/java-spiffe-helper.yaml
+      - run: kubectl wait pod/java-spiffe-helper --for condition=Ready --timeout=90s
```

Comment:
This section could benefit from added clarity through the use of descriptive step names and breaking down complex commands for better readability:

Suggestion:

```suggestion
- name: Setup Kubernetes in Docker (KIND)
  uses: helm/kind-action@v1
  with:
    cluster_name: kind

- name: Load Docker Image into KIND
  run: kind load docker-image java-spiffe-helper:test --name kind

- name: Install SPIRE CRDs via Helm
  run: helm upgrade --install -n spire-server spire-crds spire-crds --repo https://spiffe.github.io/helm-charts-hardened/ --create-namespace

- name: Configure and Install SPIRE Server
  run: |
    helm upgrade --install -n spire-server spire spire \
      --repo https://spiffe.github.io/helm-charts-hardened/ \
      -f .github/tests/spire-values.yaml \
      --set spire-server.ca_subject.common_name="$TRUSTSTORE_COMMON_NAME" \
      --set spire-server.controllerManager.identities.clusterSPIFFEIDs.java-spiffe-helper.dnsNameTemplates[0]="$KEYSTORE_COMMON_NAME"

- name: Apply java-spiffe-helper Kubernetes Configuration
  run: kubectl apply -f .github/tests/java-spiffe-helper.yaml

- name: Wait for java-spiffe-helper Pod Readiness
  run: kubectl wait pod/java-spiffe-helper --for condition=Ready --timeout=90s
```